DigitalLearningSolutions.Data/DataServices/GroupsDataService.cs

```diff
@@ -423,10 +443,35 @@ OUTPUT Inserted.GroupCustomisationId
                         (@groupId, @customisationId, @completeWithinMonths, @addedByAdminUserId, @cohortLearners, @supervisorAdminID)",
                 new
                 {
-                    groupId, customisationId, completeWithinMonths, addedByAdminUserId, cohortLearners,
-                    supervisorAdminId,
+                    groupId, customisationId, completeWithinMonths,
+                    addedByAdminUserId, cohortLearners, supervisorAdminId,
                 }
             );
         }
+
+        public void AddDelegatesWithMatchingAnswersToGroup(
+            int groupId,
+            int linkedToField,
+            int centreId,
+            string? option,
+            int? jobGroupId
+        )
+        {
+            connection.Execute(
+                @"INSERT INTO GroupDelegates (GroupID, DelegateID, AddedByFieldLink)
+                        SELECT @groupId, CandidateID, @linkedToField

```

Comment:
I'm not sure how AddedbyFieldLink is used elsewhere, but this is the SQL given in the ticket. @SteveJacksonSoft do you know if this is correct?

Reply:
AddedByFieldLink is a bit, so it should be 1 if the group is being generated from a registration field, and 0 otherwise (so probably always true for Generate Group). It's possible this works because of how SQL server runs the code, but really we want to be passing in a 1 or a 0 rather than the field number.

Reply:
Cool I've changed this method to pass in 1 rather than the linkedByField value